enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,184 @@
+
+#### ingress-vip implementation
+
+Add unicast support for the `ingress-vip` is simpler than the `api-vip`.
+
+The following steps should be done to enable unicast for `ingress-vip`:
+1. Update the Keepalived conf template files for both [control plane nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml#L49-#L73) and [compute nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/worker/00-worker/baremetal/files/baremetal-keepalived-keepalived.yaml) to include the unicast details.
+2. The keepalived-monitor container should be updated to [retrieve](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L219-#L244) the nodes (both compute and control) IP addresses from the kube-apiserver and render them to Keepalived config.
+
+### Risks and Mitigations
+
+It is important to not destabilize our existing installation method, that uses
+multicast successfully. Full parity is expected before landing.
+
+## Design Details
+
+### Test Plan
+
+Because of the fundamental inclusion of this in Baremetal IPI installations,
+any breakage by CI before landing can probably be assumed to be caused by
+this work.
+
+Take down nodes both in active as in passive mode and observe the VIP move correctly and without significantly increasing the current expected downtime in VIP changes.
+
+Force a disaster recovery scenario and see that we can recover the cluster following the usual procedure.
+
+Since the baremetal-runtimecfg is used by all on-prem platforms, we should verify that Ovirt, vSphere, and OpenStack still successfully deployed in Keepalived multicast mode.
+
+### Upgrade / Downgrade Strategy
+
+In case that the Keepalived config file was already generated by the previous SW version, Keepalived will first continue to run in a mode determined in accordance with the existing file.
+If this mode is different from the mode specified in MCO templates a Keepalived configuration migration process should be triggered.
+
+The next steps illustrate Keepalived flow in case of upgrade/downgrade:
+1. If config file exists determined_mode=mode determined from the file else determined_mode=mode specified in MCO templates
+2. Run Keepalived in determined_mode.
+3. If determined_mode != desired_mode (mode specified in MCO templates) initiate a configuratiom migration process.
+
+#### Configuration migration process
+
+Since multicast and unicast are considered as separate Keepalived domains and will not interoperate, the updated configuration should simultaneously reload in all the relevant Keepalived instances.
+
+[TBD] How to implement simultaneously reload of multiple Kepealived instances.
```

Comment:
This seems like a pretty critical design point.  I don't think this enhancement is actually "implementable" with the upgrade process still TBD

Reply:
I added a description of the suggested sync reload steps.